PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/DetectDiscouragedClientCheck.java

```diff
@@ -0,0 +1,73 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiTypeElement;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+
+/**
+ * This class extends the LocalInspectionTool to check for the use of discouraged clients
+ * in the code and suggests using other clients instead.
+ * The client data is loaded from the configuration file and the client name is checked against the
+ * discouraged client name. If the client name matches, a problem is registered with the suggestion message.
+ */
+
+public class DetectDiscouragedClientCheck extends LocalInspectionTool {
+
+    @NotNull
+    @Override
+    public JavaElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new DetectDiscouragedClientVisitor(holder, isOnTheFly);
+    }
+
+    static class DetectDiscouragedClientVisitor extends JavaElementVisitor {
+
+        // Define the fields for the visitor
+        private final boolean isOnTheFly;
+        private final ProblemsHolder holder;
+
+        // Constructor for the visitor
+        public DetectDiscouragedClientVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+            this.isOnTheFly = isOnTheFly;
+        }
+
+        private static Map<String, String> CLIENT_DATA;
+
+        // Define constants for string literals
+        private static final RuleConfig RULE_CONFIG;
+        private static final Map<String, String> CLIENTS_TO_CHECK;
```

Comment:
needs to be updated with main json loader changes

Reply:
this is up to date -- unless there's something I'm missing?